### ISSUE TITLE
Rename "Nothing" to "Something"

### ISSUE THREAD

**Issue by dsyme** [gregoryyoung/nothing#14]
With the [proposal to add Zero](https://github.com/gregoryyoung/nothing/issues/12), I suggest we rename "Nothing" to "One", or perhaps "Something".

**Comment by vossccp** [comment]
"Nothing" has an instance. Therefore, it is something. We should either make the instance private (them its really nothing at least from outside) or we should rename it to be something

**Comment by dsyme** [comment]
Yes, @gregyoung tries to make nothing, and he still ends up with something.

**Comment by gregoryyoung** [comment]
Nothing is impossible, the word itself says 'I'm possible'!
Audrey Hepburn

**Comment by dsyme** [comment]
This issue goes to the heart of differences in traditions in computing.

In the algebraic tradition, adding data to a type is multiplicative, and nothing is "one" or "unit".  So tuple types are written "int * int" with the idea that "unit * int * int" is still "int *int"

In the object oriented tradition, emptiness is nothingness is zero.  Hence the name of this repo.  In this tradition, it's unclear what "one" is.

So additive of multiplicative?  Where do you stand?
